crates/anvil/src/eth/backend/db.rs

```diff
@@ -344,3 +349,115 @@ pub struct SerializableAccountRecord {
     pub code: Bytes,
     pub storage: BTreeMap<U256, U256>,
 }
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct SerializableBlock {
+    pub header: SerializableHeader,
+    pub transactions: Vec<TypedTransaction>,
+    pub ommers: Vec<SerializableHeader>,
+}
+
+impl From<Block> for SerializableBlock {
+    fn from(block: Block) -> Self {
+        SerializableBlock {
+            header: block.header.into(),
+            transactions: block.transactions.into_iter().map(Into::into).collect(),
+            ommers: block.ommers.into_iter().map(Into::into).collect(),
+        }
+    }
+}
+
+impl From<SerializableBlock> for Block {
+    fn from(block: SerializableBlock) -> Self {
+        Block {
+            header: block.header.into(),
+            transactions: block.transactions.into_iter().map(Into::into).collect(),
+            ommers: block.ommers.into_iter().map(Into::into).collect(),
+        }
+    }
+}
+
+// TODO: only implementing here for now, need to upstream this
+//       can we make alloy_consensus::header Serializable/Deserializable?
+///      this is the same struct, literally, with derive(Serialize, Deserialize)
```

Comment:
Highlighting this. Copied a bunch of structs just to make them Serialize/Deserialize. We can clean this up by deriving Serialize/Deserialize upstream in alloy if needed.

Reply:
we should upstream this instead, looks like the serde impl for header is just missing